Client/Frontend/Browser/Helpers/ReadyStateScriptHelper.swift

```diff
@@ -0,0 +1,82 @@
+// Copyright 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import Shared
+import WebKit
+
+private let log = Logger.browserLogger
+
+struct ReadyState: Codable {
+  let securityToken: String
+  let state: State
+  
+  enum State: String, Codable {
+    // Page State
+    case loading
+    case interactive
+    case complete
+    case loaded
+    
+    // History State
+    case pushstate
+    case replacestate
+    case popstate
+  }
+  
+  public static func from(message: WKScriptMessage) -> ReadyState? {
+    if !JSONSerialization.isValidJSONObject(message.body) {
+      return nil
+    }
+
+    do {
+      let data = try JSONSerialization.data(withJSONObject: message.body, options: [])
+      return try JSONDecoder().decode(ReadyState.self, from: data)
+    } catch {
+      log.error("Error Decoding ReadyState: \(error)")
```

Comment:
@diracdeltas No it won't contain any sensitive info. It will only contain a reason as to why the JSON failed to deserialize. IE: `"Expected array, but got dictionary"`. `Expected String, but received Integer`, etc.